src/reference/00-Getting-Started/05-Basic-Def.md

```diff
@@ -2,104 +2,63 @@
-`project/` subdirectory of the base directory to define commonly used functions and values.
+This page discusses the `build.sbt` build definition.
 
 ### What is a Build Definition?
 
 After examining a set of directories and processing build definition files, sbt
-ends up with `Project` definitions.
+ends up with a set of `Project` definitions.
 
 In `build.sbt` you might create a [Project](../api/sbt/Project.html) definition of
 the project located in the current directory like this:
 
 ```scala
 lazy val root = (project in file("."))
+  .settings(
+    name := "Hello",
+    scalaVersion := "$example_scala_version$"
+  )
 ```
 
 Each project is associated with an immutable map (set of key-value pairs) describing the project.
 
 For example, one key is `name` and it maps to a string value, the name of
 your project.
-
-*Build definition files do not affect sbt's map directly.*
-
-Instead, the build definition creates a huge list of objects with type
-`Setting[T]` where `T` is the type of the value in the map. A `Setting`
-describes a *transformation to the map*, such as adding a new key-value
-pair or appending to an existing value. (In the spirit of functional
-programming with immutable data structures and values, a transformation
-returns a new map -- it does not update the old map in-place.)
-
-Here is how you associate the `Setting[String]` for the name of
-the project located in the current directory:
+The key-value pairs are listed under the `settings(...)` method as follows:
 
 ```scala
-lazy val root = (project in file(".")).
-  settings(
-    name := "hello"
+lazy val root = (project in file("."))
+  .settings(
+    name := "Hello",
+    scalaVersion := "$example_scala_version$"
   )
 ```
 
-This `Setting[String]` transforms the map by adding (or replacing) the
-name key, giving it the value `"hello"`. The transformed map becomes sbt's
-new map.
-
-To create the map, sbt first sorts the list of settings so that all
-changes to the same key are made together, and values that depend on
-other keys are processed after the keys they depend on. Then sbt walks
-over the sorted list of `Settings` and applies each one to the map in
-turn.
-
-Summary: A build definition defines `Project`s with a list of `Setting[T]`, where a
-`Setting[T]` is a transformation affecting sbt's map of key-value pairs
-and `T` is the type of each value.
-
 ### How build.sbt defines settings
 
 `build.sbt` defines a `Project`, which holds a list of Scala expressions called `settings`.
 
-Here's an example:
-
 ```scala
-lazy val commonSettings = Seq(
-  organization := "com.example",
-  version := "0.1.0",
-  scalaVersion := "$example_scala_version$"
-)
-
-lazy val root = (project in file(".")).
-  settings(commonSettings: _*).
-  settings(
-    name := "hello"
+lazy val root = (project in file("."))
+  .settings(
+    name         := "hello",
+    organization := "com.example",
+    scalaVersion := "$example_scala_version$",
+    version      := "0.1.0-SNAPSHOT"
```

Comment:
I still believe one should use ThisBuild scope as much as possible, over the commonSettings approach (for one, the state builds faster as there's less to compute). Sadly, for implementation details, you often have to fallback to commonSettings for some things (eg. anything depending on baseDirectory, including transitively dependant).

As such, I would introduce inThisBuild, and fairly early, as I agree to a more complete example that can be stripped down, instead of a minimal one that has to be built up.

Though I would not put it nested in the .settings of the root project. Firstly, it's more nesting than necessary. Secondly it makes it look like there's a specific reason why it's there, which isn't true. (I'm my fantasy sbt, settings not nested in a project are automatically scoped in ThisBuild, but alas..)